module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -404,48 +425,83 @@ public function getConfig($function, $params = null)
      * Check item location against list of configured locations
      * where holds should be offered
      *
-     * @param string $locationName locationName from getHolding
+     * @param string $locationId locationId from getHolding
      *
      * @return bool
      */
-    protected function isHoldable($locationName)
+    protected function isHoldable($locationId)
     {
         return !in_array(
-            $locationName,
+            $locationId,
```

Comment:
Is this changing the nature/meaning of the excludeHoldLocations setting in FOLIO.ini? It seems like the examples there show names rather than IDs, so if we need to change this, we should make sure that matches. Or do we need to support both scenarios somehow?

Reply:
This is a mistake on my part. It should still be based off of name. In Brad's change he sets locationName from FOLIO's discoveryDisplayName if it exists where previously it had just been from the "name" property. So Maybe I need to add both the name and discoveryDisplayName to the map of location id's and names that gets cached, and then have isHoldable check against both.

Reply:
There's a lot of potential for ambiguity and instability here, if we're matching on multiple fields, and some of those fields can potentially change over time. It does seem like it's at least worth caching and matching on the regular "name" field to maintain existing behavior. If you think it would be safe to also match on discoveryDisplayName, that might be worthwhile as an expansion... and if we're going to match on multiple fields, should we also support matching on ID, since I doubt there's any risk of collision between ID and name, and then that offers the most stable possible option for those who care to use it?

Reply:
I missed this comment before I made my last change. After consulting with Kelly at Simmons, I left the behavior of isHoldable alone, so it matches on the VuFind's location of the location Name which is either discoveryDisplayName if set or name if not. It's not great, but after looking at the UI for locations, I see that you can't save a location in the UI without setting a discoveryDisplayName. It's just that that property is not required by the backend storage module. So maybe we won't run into too much ambiguity. I don't really love this solution, and the whole feature is a bit of a hack to start with. 

Let me know what you think.

Reply:
That sounds reasonable to me. We can always revisit in another PR if it turns out that changes are needed; for now, minimizing unnecessary change/complexity seems like a wise strategy.